packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts

```diff
@@ -86,6 +88,26 @@ export class FargateService extends BaseService implements IFargateService {
       throw new Error('A TaskDefinition must have at least one essential container');
     }
   }
+
+  /**
+   * Adds AWS X-Ray daemon as a sidecar container to enable tracing
+   */
+  public addTracing(props: TracingOptions = {}) {
+    // TODO: adjust task size based on container-level resources (cpu/memory)?
+    const optIn = props.enableLogging !== undefined ? props.enableLogging : true;
+    const xray = this.taskDefinition.addContainer("xray-daemon", {
+      image: ContainerImage.fromRegistry("amazon/aws-xray-daemon"),
+      cpu: props.cpu,
+      memoryReservationMiB: props.memoryReservationMiB,
+      essential: props.essential || false,
+      logging: optIn ? new AwsLogDriver({ streamPrefix: this.node.id}) : undefined
```

Comment:
Perhaps `node.path` or `node.unique` is a better choice

Reply:
Using this.node.uniqueId instead of this.node.id